packages/@o3r/schematics/src/utility/matching-peers.ts

```diff
@@ -19,6 +19,7 @@ export function getPeerDepWithPattern(packageJsonPath: string, pattern: RegExp |
   const matchingPackagesVersions = Object.fromEntries(
     Object.entries(packageJsonContent.peerDependencies || {})
       .filter(([peerDep]) => (Array.isArray(pattern) ? pattern.includes(peerDep) : pattern.test(peerDep)) && !optionalPackages.includes(peerDep))
+      .map(([peerDep, range]) => ([peerDep, packageJsonContent.generatorDependencies?.[peerDep] || range]))
```

Comment:
__Q__: Why are you taking the range from the generator dependencies ? It could be different from the one defined in 'peerDeps'? I mean in peerDeps we'll have '^' and in generator deps we'll stay with '~'?

Reply:
We have the case already and yes, we can use `generatorDependencies` to specify version within the `peerDependencies` ranges